### ISSUE TITLE
Context object might be null

### ISSUE THREAD

**Issue by albertogasparin** [Ephem/react#22]
First of all, thanks for making this easy to test Suspense in SSR. I've tried running a big application with it and hit some edge cases.
It looks like that `context` can be a `null` object, so the methods expecting on it to be an object throw. Didn't have time to investigate why context was `null` in the first place (my guess is maybe a bug in `popProvider`) but fixed it by checking for `null` where context was expected:

```js
function ReactDOMServerRendererAsync(children, makeStaticMarkup, renderContext) {
      // ...
      for (var i = 0; i < _this.contextStack.length; i += 1) {
        if (!_this.contextStack[i]) { // <- contextStack[I] can be null
          _this.contextStack[i] = {};
        }
        _this.contextStack[i][_this.threadID] = renderContext.contextValues[i];
      }
      // ...
}
```
And also

```js
ReactDOMServerRendererAsync.prototype.getClonedRenderContext = function getClonedRenderContext() {
    var _this2 = this;

    var contextValues = this.contextStack.map(function (context) {
      return context && context[_this2.threadID]; // <- here context can be null
    });
    // ...
}
```

**Comment by Ephem** [comment]
Great find and solution looks good! I'm a bit embarrassed I didn't code more defensively there.. 😁

I'll try and fix this and push a patch today.

**Comment by Ephem** [comment]
Btw, I think the reason this happens is:

Any time React runs into a Provider, that Providers context is pushed to `contextStack`.
When traversing back up the tree and a Provider is closed, the last value of that stack is set to null, at least until another Provider overrides it again, which means null is a perfectly fine value.

I think something like this would reproduce the error:

```jsx
renderToStringAsync(
  <Suspense fallback="">
    <ContextProvider>
      <NestedComponent />
    </ContextProvider>
    <ComponentThatSuspends />
  </Suspense>
);
```

In this case the last value of the `contextStack` would be null.

**Comment by Ephem** [comment]
Fixed in `0.2.2`
